core/trino-main/src/main/java/io/trino/sql/planner/SymbolKeyDeserializer.java

```diff
@@ -33,7 +33,10 @@ public SymbolKeyDeserializer(TypeManager typeManager)
     @Override
     public Object deserializeKey(String key, DeserializationContext context)
     {
-        String[] parts = key.split("::");
```

Comment:
can we just reverse the order of serialization?

we know that type doesn't contain a colon so, `type:name`
which also makes the return look like `return new Symbol(typeManager.getType(TypeId.of(parts[0])), parts[1]);` instead of parts[1], parts[0]

Reply:
Actually, the current implementation is not enough. Types may contain `::` in row field names. I need to implement something more robust.

Reply:
base64(typeId) + separator + base64(columnName)

Reply:
or at least base64(typeid) + separator + columnName ?

Reply:
Or can you just sanitize symbols, and strip `:` of it?